persistence-cassandra/javadsl/src/test/scala/com/lightbend/lagom/javadsl/persistence/cassandra/CassandraPersistenceSpec.scala

```diff
@@ -34,13 +37,19 @@ class CassandraPersistenceSpec(system: ActorSystem) extends ActorSystemSpec(syst
   override def beforeAll(): Unit = {
     super.beforeAll()
 
+    // Join ourselves - needed because the Cassandra offset store uses cluster startup task
+    val cluster = Cluster(system)
+    cluster.join(cluster.selfAddress)
```

Comment:
I don't understand why this would be needed? It must be better to have Cassandra running as early as possible so that any things in the test or Lagom itself can connect if they startup as part of the cluster formation.

Reply:
Both write and read sides depend on cluter and therefore it's ok to start the cluter first.

Reply:
my point is that if they start before Cassandra is started the initial connect to Cassandra may fail. That may cause other problems.